pkg/node/bootstrap.go

```diff
@@ -202,61 +203,55 @@ func bootstrapNode(
 		return nil, errors.New("timed out waiting for kademlia peers")
 	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
-	defer cancel()
+	fetch := func() (*postage.ChainSnapshot, error) {
+		ctx, cancel := context.WithTimeout(context.Background(), timeout)
+		defer cancel()
 
-	logger.Info("bootstrap: trying to fetch stamps snapshot")
+		logger.Info("bootstrap: trying to fetch stamps snapshot")
 
-	var (
-		snapshotReference swarm.Address
-		reader            file.Joiner
-		l                 int64
-		eventsJSON        []byte
-	)
-
-	for i := 0; i < getSnapshotRetries; i++ {
-		snapshotReference, err = getLatestSnapshot(ctx, ns, snapshotFeed)
+		snapshotReference, err := getLatestSnapshot(ctx, ns, snapshotFeed)
```

Comment:
If we got past this stage in the first try, we should ideally not do this again. The reference will be the same and there might be more lookups triggered because of this.

Maybe we can separate these two operations and retry each one separately if they fail.